scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/JobEmailNotification.java

```diff
@@ -122,17 +149,19 @@ public boolean doCheckAndSend() throws JobEmailNotificationException {
         }
     }
 
-    public void checkAndSendAsync() {
+    public void checkAndSendAsync(boolean withAttachment) {
         this.asyncMailSender.submit(() -> {
             try {
-                boolean sent = doCheckAndSend();
+                boolean sent = doCheckAndSend(withAttachment);
                 if (sent) {
                     jlogger.info(jobState.getId(), "sent notification email for finished job to " + getTo());
                 }
             } catch (JobEmailNotificationException e) {
                 jlogger.warn(jobState.getId(),
                              "failed to send email notification to " + e.getEmailTarget() + ": " + e.getMessage());
                 logger.trace("Stack trace:", e);
+            } catch (Exception e) {
```

Comment:
please add jlogger.warn(jobState.getId(),
                             "failed to send email notification to " + e.getEmailTarget() + ": " + e.getMessage());	                             "failed to send email notification to " + e.getEmailTarget() + ": " + e.getMessage());

Reply:
I can not add this, because e doesn't have the method e.getMessage(), it is a method of JobEmailNotificationException, so I separate these two.

Reply:
you mean e does not have e.getEmailTarget(), indeed.

But you can simply do:

jlogger.warn(jobState.getId(),
                             "failed to send email notification;" + e.getMessage());	                             "failed to send email notification: " + e.getMessage());

Reply:
it's done, thanks